lib/galaxy/jobs/__init__.py

```diff
@@ -1228,7 +1238,7 @@ def working_directory(self):
 
     def working_directory_exists(self):
         job = self.get_job()
-        return self.app.object_store.exists(job, base_dir='job_work', dir_only=True, obj_dir=True)
```

Comment:
from lib/galaxy/objectstore/__init__.py
```
class ObjectStore
...
def exists(self, obj, base_dir=None, dir_only=False, extra_dir=None, extra_dir_at_root=False, alt_name=None):
```

does not have `obj_dir`

Reply:
That's the abstractmethod, but the implementation accepts **kwargs and defers to `_exists`, which calls `_construct_path` which does accept `obj_dir`. I think the fix here is to change the abstractmethod's signature